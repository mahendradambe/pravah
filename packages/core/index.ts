import { Container } from "./src/container";
import { Node } from "./src/node";

export const container = new Container();

container.registerNode(Node);

export * from "./src/connection";
export * from "./src/container";
export * from "./src/helpers";
export * from "./src/node";
export * from "./src/param-port";
export * from "./src/port";
export * from "./src/procedure";
export * from "./src/result-port";
export * from "./src/types";

