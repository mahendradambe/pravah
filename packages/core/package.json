{
    "name": "@pravah/core",
    "version": "0.0.1-prerelease",
    "module": "src/index.ts",
    "main": "dist/index.js",
    "types": "dist/index.d.ts",
    "scripts": {
        "build": "bun build.ts"
    },
    "author": "Mahendra Dambe",
    "license": "MIT",
    "peerDependencies": {
        "@pravah/stream": "workspace:*",
        "@pravah/utils": "workspace:*"
    },
    "devDependencies": {
        "@types/bun": "1.0.11",
        "@pravah/build": "workspace:*"
    }
}
