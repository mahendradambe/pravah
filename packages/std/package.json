{
    "name": "@pravah/std",
    "version": "0.0.1-prerelease",
    "module": "src/index.ts",
    "main": "dist/index.js",
    "types": "dist/index.d.ts",
    "author": "Mahendra Dambe",
    "scripts": {
        "build": "bun build.ts"
    },
    "license": "MIT",
    "peerDependencies": {
        "@pravah/core": "workspace:*",
        "@pravah/stream": "workspace:*",
        "pupa": "^3.1.0"
    },
    "devDependencies": {
        "@types/bun": "1.0.11"
    }
}
