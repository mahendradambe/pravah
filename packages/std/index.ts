import * as Core from "@pravah/core";
import * as Def from "./src/definitions";
import * as Type from "./src/types";

const container = new Core.Container();

container.extend(Core.container);
container.registerDefinition(Def.LogDef);
container.registerDefinition(Def.SumDef);
container.registerDefinition(Def.TemplateDef);
container.registerType(Type.BooleanType);
container.registerType(Type.NumberType);
container.registerType(Type.StringType);

export * from "./src/definitions";
export { Def, Type, container };

