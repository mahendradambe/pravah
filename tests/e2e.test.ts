import * as Core from "@pravah/core";
import * as Std from "@pravah/std";
import { logsum } from "./mock";

describe("e2e", () => {
    it("works", () => {
        const container = Std.container;

        const procedure = new Core.Procedure(container, logsum.id, logsum);
        //     __________________
        //  *  [(0)             ]
        //  *  |     sumA  (sum)|\
        //  *  [(1)             ] \     ______________
        //  *  [________________]  \    |            |
        //  *                      | -> [(1)  log    ]
        //  *  __________________  /    |____________|
        //  *  [(0)             ] /
        //  *  |     sumB  (sum)|/
        //  *  [(1)             ]
        //  *  [________________]
        //  *  */

        procedure.start();
        procedure.activate();

        const procedureData = procedure.serialize();

        expect(JSON.stringify(procedureData, null, 4)).toMatchSnapshot();
    });
});
