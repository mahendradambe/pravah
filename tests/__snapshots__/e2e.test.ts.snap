// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e works 1`] = `
"{
    "id": "logsum",
    "connections": [
        {
            "id": "sumC:0-sumA:sum",
            "param": 0,
            "result": "sum",
            "source": "sumA",
            "target": "sumC",
            "meta": {}
        },
        {
            "id": "sumC:1-sumB:sum",
            "param": 1,
            "result": "sum",
            "source": "sumB",
            "target": "sumC",
            "meta": {}
        },
        {
            "id": "log:0-sumC:sum",
            "param": 0,
            "result": "sum",
            "source": "sumC",
            "target": "log",
            "meta": {}
        }
    ],
    "nodes": [
        {
            "id": "sumA",
            "name": "std:def:sum",
            "type": "core:node",
            "params": [
                {
                    "type": "std:type:number",
                    "order": 0,
                    "value": 1
                },
                {
                    "type": "std:type:number",
                    "order": 1,
                    "value": 2
                }
            ],
            "results": [
                {
                    "alias": "result",
                    "type": "std:type:number"
                },
                {
                    "alias": "sum",
                    "type": "std:type:number"
                }
            ],
            "returnType": "std:type:number",
            "cache": 3,
            "meta": {}
        },
        {
            "id": "sumB",
            "name": "std:def:sum",
            "type": "core:node",
            "params": [
                {
                    "type": "std:type:number",
                    "order": 0,
                    "value": 3
                },
                {
                    "type": "std:type:number",
                    "order": 1,
                    "value": 4
                }
            ],
            "results": [
                {
                    "alias": "result",
                    "type": "std:type:number"
                },
                {
                    "alias": "sum",
                    "type": "std:type:number"
                }
            ],
            "returnType": "std:type:number",
            "cache": 7,
            "meta": {}
        },
        {
            "id": "sumC",
            "name": "std:def:sum",
            "type": "core:node",
            "params": [
                {
                    "type": "std:type:number",
                    "order": 0,
                    "value": 3
                },
                {
                    "type": "std:type:number",
                    "order": 1,
                    "value": 7
                }
            ],
            "results": [
                {
                    "alias": "result",
                    "type": "std:type:number"
                },
                {
                    "alias": "sum",
                    "type": "std:type:number"
                }
            ],
            "returnType": "std:type:number",
            "cache": 10,
            "meta": {}
        },
        {
            "id": "log",
            "name": "std:def:log",
            "type": "core:node",
            "params": [
                {
                    "type": "std:type:number",
                    "order": 0,
                    "value": 10
                }
            ],
            "results": [],
            "meta": {}
        }
    ],
    "meta": {}
}"
`;
